src/code/z_player_lib.c

```diff
@@ -1,153 +1,1043 @@
+    }
+
+    func_80123DA4(player);
+}
+
+void func_80123E90(PlayState* play, Actor* actor) {
+    Player* player = GET_PLAYER(play);
+
+    func_80123DC0(player);
+    player->unk_730 = actor;
+    player->unk_A78 = actor;
+    player->stateFlags1 |= PLAYER_STATE1_10000;
+    func_800DFD78(Play_GetCamera(play, CAM_ID_MAIN), 8, actor);
+    Camera_ChangeMode(Play_GetCamera(play, CAM_ID_MAIN), 9);
+}
+
+s32 func_80123F14(PlayState* play) {
+    Player* player = GET_PLAYER(play);
+
+    return player->stateFlags1 & PLAYER_STATE1_800000;
+}
+
+s32 func_80123F2C(PlayState* play, s32 ammo) {
+    play->unk_1887C = ammo + 1;
+
+    return 1;
+}
+
+s32 Player_IsBurningStickInRange(PlayState* play, Vec3f* pos, f32 xzRange, f32 yRange) {
+    Player* this = GET_PLAYER(play);
+    Vec3f diff;
+    s32 pad;
+
+    if ((this->itemActionParam == PLAYER_AP_STICK) && (this->unk_B28 != 0)) {
+        Math_Vec3f_Diff(&this->meleeWeaponInfo[0].tip, pos, &diff);
+        return (SQXZ(diff) <= SQ(xzRange)) && (0.0f <= diff.y) && (diff.y <= yRange);
+    }
+
+    return false;
+}
+
+u8 Player_GetStrength(void) {
+    return sPlayerStrengths[(void)0, gSaveContext.save.playerForm];
+}
+
+u8 Player_GetMask(PlayState* play) {
+    Player* player = GET_PLAYER(play);
+
+    return player->currentMask;
+}
+
+// Unused
+void Player_RemoveMask(PlayState* play) {
+    Player* player = GET_PLAYER(play);
+
+    player->currentMask = PLAYER_MASK_NONE;
+}
+
+s32 Player_HasMirrorShieldEquipped(PlayState* play) {
+    Player* player = GET_PLAYER(play);
+
+    return (player->transformation == PLAYER_FORM_HUMAN) && (player->currentShield == PLAYER_SHIELD_MIRROR_SHIELD);
+}
+
+s32 Player_IsHoldingMirrorShield(PlayState* play) {
+    Player* player = GET_PLAYER(play);
+
+    return (player->transformation == PLAYER_FORM_HUMAN) && (player->rightHandType == 8) &&
+           (player->currentShield == PLAYER_SHIELD_MIRROR_SHIELD);
+}
+
+s32 Player_IsHoldingHookshot(Player* player) {
+    return player->itemActionParam == PLAYER_AP_HOOKSHOT;
+}
+
+s32 func_801240DC(Player* player) {
+    return Player_IsHoldingHookshot(player) && (player->heldActor == NULL);
+}
+
+s32 func_80124110(Player* player, s32 actionParam) {
+    s32 temp_v0 = actionParam - PLAYER_AP_UNK_2;
+
+    if (player->transformation != PLAYER_FORM_GORON) {
+        if (((actionParam - PLAYER_AP_UNK_2) > (PLAYER_AP_UNK_2 - PLAYER_AP_UNK_2)) &&
+            ((actionParam - PLAYER_AP_UNK_2) < (PLAYER_AP_SWORD_GREAT_FAIRY - PLAYER_AP_UNK_2))) {
+            return temp_v0;
+        }
+    }
+
+    return 0;
+}
+
+s32 func_80124148(Player* player) {
+    return func_80124110(player, player->itemActionParam);
+}
+
+s32 Player_ActionToMeleeWeapon(s32 actionParam) {
+    s32 weapon = actionParam - PLAYER_AP_UNK_2;
+
+    if ((weapon > (PLAYER_AP_UNK_2 - PLAYER_AP_UNK_2)) && (weapon <= (PLAYER_AP_UNK_8 - PLAYER_AP_UNK_2))) {
+        return weapon;
+    }
+    return 0;
+}
```

Comment:
Did a quick rummage in the item to action param function, UNK_2 is the fishing rod, as in OoT. So Kokiri Sword - 1 is probably a better thing to use. UNK_8 is zora fins, so I suppose this counts it as a sword?

Reply:
I guess zora fins being a melee weapon (not sword) kinda makes sense.
The fishing rod does not count as a two handed melee weapon?

Reply:
Well, the point is that it returns 0 for everything else, right? I don't think it's coercing every other value into a fishing rod.